src/views/MainPage/components/Introduce/MotionTitle/index.tsx

```diff
@@ -0,0 +1,55 @@
+import { useScroll, useTransform } from 'framer-motion';
+import { RefObject, useEffect, useState } from 'react';
+import { useIsMobile } from '@src/hooks/useDevice';
+import * as S from './style';
+
+interface MotionTitleProps {
+  contentRef: RefObject<HTMLElement>;
+  content: string;
+}
+
+export default function MotionTitle({ contentRef, content }: MotionTitleProps) {
+  const isMobileSize = useIsMobile('768px');
+
+  const { scrollYProgress } = useScroll({
+    target: contentRef,
+    offset: ['end center', 'start start'],
+  });
+
+  const [clipPathValue, setClipPathValue] = useState<string>('inset(0% 0% 0% 0%)');
+  const [opacityValue, setOpacityValue] = useState<number>(0);
+  const scrollValue = useTransform(scrollYProgress, [1, 0.4], ['100%', '0%']);
+
+  useEffect(() => {
+    const unsubscribe = scrollValue.on('change', (value) => {
+      const percentValue = Number(value.split('%')[0]);
+      isMobileSize
+        ? setOpacityValue((100 - percentValue) / 100)
+        : setClipPathValue(`inset(0% ${percentValue}% 0% 0%)`);
+    });
```

Comment:
지금 상태를 두 개 사용하고 있는데 style 상태를 하나 만들어도 될 것 같아요

```suggestion
    const unsubscribe = scrollValue.on('change', (value) => {
      const percentValue = Number(value.split('%')[0]);
      const newStyle = isMobileSize
        ? { opacity: (100 - percentValue) / 100 }
        : { clipPath: `inset(0% ${percentValue}% 0% 0%)` };
      setStyle(newStyle);
    });
```

이렇게 하면 조건부 렌더링 없이 다음처럼 작성할 수 있을 것 같습니다..!!
`<S.MotionTitle style={style} data-text={content} />`

Reply:
두개의 스타일을 분리하려고만 하다보니 하나의 스타일로 합치는걸 생각못했네요...! 좋은 의견 감사해요!!